apiserver/v1/types_deprecatedapirequest.go

```diff
@@ -0,0 +1,110 @@
+// Package v1 is an api version in the apiserver.openshift.io group
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Cluster"
+// +kubebuilder:subresource:status
+// +genclient:nonNamespaced
+
+// DeprecatedAPIRequest tracts requests made to a deprecated API. The instance name should
+// be of the form `resource.version.group`, matching the deprecated resource.
+type DeprecatedAPIRequest struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec defines the characteristics of the resource.
+	// +kubebuilder:validation:Required
+	// +required
+	Spec DeprecatedAPIRequestSpec `json:"spec"`
+
+	// Status contains the observed state of the resource.
+	Status DeprecatedAPIRequestStatus `json:"status,omitempty"`
+}
+
+type DeprecatedAPIRequestSpec struct {
+	// RemovedRelease is when the API will be removed.
+	// +kubebuilder:validation:MaxLength=64
+	// +required
+	RemovedRelease string `json:"removedRelease"`
```

Comment:
who sets this?

Reply:
Would expect status only

Reply:
This field was requested by @deads2k. Currently, the controller that logs these requests fills in this value when creating the initial cr instance.

Reply:
I can get onboard with that. Would like to see minLength and a pattern (something like digit.digit+)

Reply:
Added `	// +kubebuilder:validation:Pattern=^[0-9][0-9]*\.[0-9][0-9]*$`